sender.go

```diff
@@ -52,9 +62,34 @@ var gcmSendEndpoint = GcmSendEndpoint
 //	}
 type Sender struct {
 	ApiKey string
+	URL    string
 	Http   *http.Client
 }
 
+// NewClient returns a new sender with the given URL and apiKey.
+// If one of input is empty or URL is malformed, returns error.
+// It sets http.DefaultHTTP client for http connection to server.
+// If you need our own configuration overwrite it.
+func NewClient(urlString, apiKey string) (*Sender, error) {
+	if len(urlString) == 0 {
+		return nil, fmt.Errorf("missing GCM/FCM endpoint url")
+	}
+
+	if len(apiKey) == 0 {
+		return nil, fmt.Errorf("missing API Key")
+	}
+
+	if _, err := url.Parse(urlString); err != nil {
+		return nil, fmt.Errorf("failed to parse URL %q: %s", urlString, err)
+	}
+
+	return &Sender{
+		URL:    urlString,
+		ApiKey: apiKey,
+		Http:   http.DefaultClient,
```

Comment:
IMO, using http.DefaultClient is bad approach. Because http.DefaultClient.Timeout is zero by default. It means no timeout and it should not set in production server.